examples/rust/udp-echo.rs

```diff
@@ -230,10 +231,10 @@ impl Application {
 
     #[cfg(target_os = "windows")]
     /// Converts a [sockaddr] into a [SocketAddrV4].
-    pub fn sockaddr_to_socketaddrv4(saddr: *const SOCKADDR_IN) -> Result<SocketAddrV4> {
+    pub fn sockaddr_to_socketaddrv4(saddr: *const SOCKADDR) -> Result<SocketAddrV4> {
         // TODO: Change the logic below and rename this function once we support V6 addresses as well.
 
-        let sin: SOCKADDR_IN = unsafe { *saddr };
+        let sin: SOCKADDR_IN = unsafe { *mem::transmute::<*const SOCKADDR, *const SOCKADDR_IN>(saddr) };
```

Comment:
Why dropping the `mem::transmute` statement?

Rationale for keeping this: `mem::transmute` provides a slightly stronger typing (https://doc.rust-lang.org/nomicon/transmutes.html)